### PR TITLE
#2040: replace immutable HashSet by Set.of() to reduce code

### LINKED ISSUE #2040: Use Set.of() instead of new HashSet<>(Arrays.asList()) in MetadataConstraints.
Java recently introduced the Set.of() static methods for creating immutable sets.  These new methods could be used [here](https://github.com/apache/accumulo/blob/9afab2c1b931d100ed6bad5cfe028c28da944427/server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java#L79) and [here](https://github.com/apache/accumulo/blob/9afab2c1b931d100ed6bad5cfe028c28da944427/server/base/src/main/java/org/apache/accumulo/server/constraints/MetadataConstraints.java#L90) instead of `new HashSet<>(Arrays.asList(`. This would shorten the code and make the sets immutable, which seems to be the intention of the code.

### PR DESCRIPTION
**Pull request by hguercan**
replace immutable HashSet usages by Set.of() to reduce code 
* adjusted imports
* closes #2040

### CODE REVIEW

**Review by EdColeman — COMMENTED** [review]
The changes look good - I'm running tests now.  Did you happen to check other places where the substitution may be possible?

Currently, the following files match as possible candidates, but I've not looked further.
`
./core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java
./server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java
./server/base/src/main/java/org/apache/accumulo/server/security/handler/KerberosAuthenticator.java
./server/base/src/main/java/org/apache/accumulo/server/util/FileSystemMonitor.java
./server/manager/src/test/java/org/apache/accumulo/manager/replication/UnorderedWorkAssignerTest.java
./server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java
./shell/src/main/java/org/apache/accumulo/shell/commands/DUCommand.java
./shell/src/test/java/org/apache/accumulo/shell/commands/DeleteTableCommandTest.java
./test/src/main/java/org/apache/accumulo/test/functional/ConfigurableCompactionIT.java
./test/src/main/java/org/apache/accumulo/test/functional/ManagerMetricsIT.java
./test/src/main/java/org/apache/accumulo/test/functional/PermissionsIT.java
./test/src/main/java/org/apache/accumulo/test/functional/SummaryIT.java
./test/src/main/java/org/apache/accumulo/test/OrIteratorIT.java
./test/src/main/java/org/apache/accumulo/test/ShellServerIT.java
./test/src/main/java/org/apache/accumulo/test/TableOperationsIT.java
./test/src/main/java/org/apache/accumulo/test/UserCompactionStrategyIT.java

`

**Comment by hguercan** [comment]
Will check the other files too, probably was using your proposed command not on project root level

**Comment by EdColeman** [comment]
Thanks!  Was hoping to avoid work duplication if you had.

**Review by EdColeman — CHANGES_REQUESTED** [review]
@hguercan is going to see if there are other candidates for replacement.

**Comment by hguercan** [comment]
Also added Locale setting in NumUtilTest to US since unit testing fails if LC settings for numbers are set to DE for example

 accumulo git:(replace_hashset_by_setof) ✗ find . -name "*.java" | xargs grep asList | grep new | grep Set

./core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java: Sets.copyOf but wasnt sure about null values
./test/src/main/java/org/apache/accumulo/test/UserCompactionStrategyIT.java: SortedSet --> TreeSet
./test/src/main/java/org/apache/accumulo/test/OrIteratorIT.java: TreeSet
./test/src/main/java/org/apache/accumulo/test/OrIteratorIT.java: TreeSet
./test/src/main/java/org/apache/accumulo/test/TableOperationsIT.java:  TreeSet
./test/src/main/java/org/apache/accumulo/test/functional/ConfigurableCompactionIT.java: TreeSet
./test/src/main/java/org/apache/accumulo/test/functional/SummaryIT.java: TreeSet
./shell/src/main/java/org/apache/accumulo/shell/commands/DUCommand.java: SortedSet --> TreeSet
./server/tserver/src/main/java/org/apache/accumulo/tserver/AssignmentHandler.java:  TreeSet
./server/manager/src/test/java/org/apache/accumulo/manager/replication/UnorderedWorkAssignerTest.java: gets modified
./server/base/src/main/java/org/apache/accumulo/server/rpc/TServerUtils.java: gets modified

**Review by EdColeman — COMMENTED** [review]
Overall the changes look good.  Thanks for all of the work.  Running the sunnyDay tests I do get a failure in ShellServerIT - but I believe that is unrelated to any of these changes.  I going to see if I can track that failure down and allow @keith-turner to weigh in if he wants.

File: core/src/test/java/org/apache/accumulo/core/util/NumUtilTest.java [hunk 1]
```diff
@@ -22,12 +22,15 @@
...
 
+import java.util.Locale;
+
 import org.junit.Test;
 
 public class NumUtilTest {
 
   @Test
   public void testBigNumberForSize() {
+    Locale.setDefault(Locale.US);
```
Comment (ctubbsii) [inline core/src/test/java/org/apache/accumulo/core/util/NumUtilTest.java:33]: This change seems unrelated. Is it necessary? If it's an issue, it could be a separate PR.

**Review by ctubbsii — APPROVED** [review]
Looks good to me, provided the tests pass. I did have the one question about the necessity of the Locale.US change, but it's not a problem to include it, I think.

**Comment by hguercan** [comment]
Thx for the reviews. Have to admit with the Locale.US addition I also was not sure if it belongs to "this" issue or should be addressed in a dedicated PR but thought unit tests needs to be independent as possible and this setting would be needed to get there and also to verify my work for this issue. Would be totally ok to revert and address it in a dedicated PR if you think differently.

File: core/src/test/java/org/apache/accumulo/core/util/NumUtilTest.java (hunk 1 above)
Comment (EdColeman) [reply core/src/test/java/org/apache/accumulo/core/util/NumUtilTest.java:33]: I created https://github.com/apache/accumulo/pull/2072 - that will separate this issue into its own commit. Once that change is merged, this can be re-based and merged.

**Comment by EdColeman** [comment]
Thanks for the contribution @hguercan  and welcome to the Apache Accumulo community. If you would like to be listed as a contributor on Accumulo's people page please make a website PR for people.md. https://github.com/apache/accumulo-website/edit/main/pages/people.md
If you intend to make more contributions, please consider subscribing to the dev list and introducing yourself. Also, you can use the dev list or slack to reach out if you have questions about anything. https://accumulo.apache.org/contact-us/
